functions/src/index.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+exports.__esModule = true;
+var functions = require("firebase-functions");
+var axios_1 = require("axios");
+var cors = require('cors')({ origin: true });
+var key = functions.config().api.key;
+function compressImage(url) {
+    var options = {
+        url: '/shrink',
+        method: 'post',
+        baseURL: 'https://api.tinify.com',
+        data: {
+            'source': { 'url': url }
+        },
+        auth: {
+            username: 'api',
+            password: key
+        },
+        headers: { 'Content-Type': 'application/json' },
+        responseType: 'json'
+    };
+    return axios_1["default"].request(options).then(function (response) {
```

Comment:
This could be refactored using our axios instance:

```javascript
return http.post('/shrink', {
    source: { url }
});
```

No need for an option object anymore.